pkg/secrets/aws/secrets.go

```diff
@@ -45,12 +52,20 @@ func (sp *SecretsProvider) ResolveSecrets(ctx context.Context, vars []string) ([
 		kv := strings.Split(env, "=")
 		key, value := kv[0], kv[1]
 		if strings.HasPrefix(value, "arn:aws:secretsmanager") {
+			vals := strings.Split(value, "#")
+			value = vals[0] // secret ARN
+			k := vals[len(vals)-1] // key to extract from SecretString
 			// get secret value
 			secret, err := sp.sm.GetSecretValue(&secretsmanager.GetSecretValueInput{SecretId: &value})
 			if err != nil {
 				return vars, errors.Wrap(err, "failed to get secret from AWS Secrets Manager")
 			}
-			env = key + "=" + *secret.SecretString
+			if isJSON(*secret.SecretString) && k != value {
+				v := gjson.Get(*secret.SecretString, k)
```

Comment:
What if `v` is null?

Reply:
Your application probably won't work since the value you're anticipating being there isn't there. I don't think this should be graceful. If I am deploying a service with a value from AWS secrets manager to be injected into my env at runtime and it deploys "successfully" but the env var was never fetched and injected, I'll end up down a rabbit hole. I'd rather be in a CrashLoopBackoff state.. Maybe the error handling should say that the key you're trying to fetch isn't there and provide some sort of direction?